scripts/start.js

```diff
@@ -150,13 +153,56 @@ function openBrowser(port) {
   opn('http://localhost:' + port + '/');
 }
 
+function addMiddleware(devServer) {
+  // `devProxy` lets you to specify a fallback server during development.
+  // Every unrecognized request will be forwarded to it.
+  var devProxy = require(paths.appPackageJson).devProxy;
+  devServer.use(historyApiFallback({
+    // For single page apps, we generally want to fallback to /index.html.
+    // However we also want to respect `devProxy` for API calls.
+    // So if `devProxy` is specified, we need to decide which fallback to use.
+    // We use a heuristic: if request `accept`s text/html, we pick /index.html.
+    // Modern browsers include text/html into `accept` header when navigating.
+    // However API calls like `fetch()` won’t generally won’t accept text/html.
+    // If this heuristic doesn’t work well for you, don’t use `devProxy`.
+    htmlAcceptHeaders: devProxy ?
+      ['text/html'] :
+      ['text/html', '*/*']
+  }));
+  if (devProxy) {
+    if (typeof devProxy !== 'string') {
+      console.log(chalk.red('When specified, "devProxy" in package.json must be a string.'));
+      console.log(chalk.red('Instead, the type of "devProxy" was "' + typeof devProxy + '".'));
+      console.log(chalk.red('Either remove "devProxy" from package.json, or make it a string.'));
+      process.exit(1);
+    }
+
+    // Otherwise, if devProxy is specified, we will let it handle any request
+    // that isn't /index.html or /sockjs-node/* (used for hot reloading).
```

Comment:
Do we let the proxy handle a request for `/`? I think the request from a browser will not actually be for `/index.html` most of the time, it will be for `/`, which makes some of these comments a bit confusing. I think the behavior is actually the same between `/index.html` and `/` though.